lib/smooth/include/smooth/application/io/spi/BME280Core.h

```diff
@@ -0,0 +1,233 @@
+            /// \param config_data Data read from register 0xF5
+            /// \param mode The mode configuration setting will be written to this address
+            /// \param humidity The humidity configuration setting will be written to this address
+            /// \param pressure The pressure configuration setting will be written to this address
+            /// \param temperature The temperature configuration setting will be written to this address
+            /// \param standby The standby times configuration setting will be written to this address
+            /// \param filter_coeff The filter coefficient configuration setting will be written to this address
+            /// \param spi_interface The spi_interface configuration setting will be written to this address
+            void get_configuration_settings(uint8_t ctrl_hum_data,
+                                            uint8_t ctrl_meas_data,
+                                            uint8_t config_data,
+                                            SensorMode& mode,
+                                            OverSampling& humidity,
+                                            OverSampling& pressure,
+                                            OverSampling& temperature,
+                                            StandbyTimeMS& standby,
+                                            FilterCoeff& filter,
+                                            SpiInterface& spi_interface);
+
+            /// Get the compensated measurements
+            /// \param measurment_data A container that holds the data read from measurement registers 0xF7..0xFE
+            /// \param humidity Compensated Humidity, in %RH
+            /// \param pressure Compenstated Pressure, in Pa
+            /// \param temperature Compensated Temperature in degree Celsius
+            //void get_measurements(std::array<uint8_t, 8>& measurement_data,
+            void get_measurements(core::util::FixedBuffer<uint8_t, 8>& measurement_data,
+                                  float& humidity,
+                                  float& pressure,
+                                  float& temperature);
+
+            /// Populate the trimming registers for later use when calculating the compensated temp, press and hum
+            // values
+            /// \param calibration_data A container that holds the data read from calibration registers 0x88..0x9F,
+            // 0xA1, 0xE1..0xE7.
+            void populate_trimming_registers(core::util::FixedBuffer<uint8_t, 32>& calibration_data);
+
+        private:
+            // These calculation methods are based on those in the datasheet at
+            // https://ae-bst.resource.bosch.com/media/_tech/media/datasheets/BST-BME280_DS001-11.pdf
+            // as of 2017-08-20. The revision of the code is 1.1 according to that document.
+            typedef int32_t BME280_S32_t;
+            typedef uint32_t BME280_U32_t;
+            typedef int64_t BME280_S64_t;
```

Comment:
Perhaps originating from my SPI-code, but lets use C++17's using instead of typedef for these three.

example: `using BME280_S32_t = int32_t `